### PR TITLE
Improve error handling of loadByResourceLocator

### PR DESCRIPTION
**Pull request by mamazu**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes nothing
| License | MIT

#### What's in this PR?
Better types

#### Why?
I am trying to understand how this works and right typing would help.

### CODE REVIEW

**Comment by alexander-schranz** [comment]
We should target `2.4` for this. @wachterjohannes can you confirm that the new exception also makes from your point of view sense?

File: src/Sulu/Component/Content/Types/ResourceLocator/Mapper/PhpcrMapper.php [hunk 1]
```diff
@@ -276,10 +276,15 @@ public function loadByResourceLocator($resourceLocator, $webspaceKey, $languageC
...
-                throw new ResourceLocatorMovedException(
-                    $this->getResourceLocator($realPath->getPath(), $webspaceKey, $languageCode, $segmentKey),
-                    $realPath->getIdentifier()
-                );
+                $locator = $this->getResourceLocator($realPath->getPath(), $webspaceKey, $languageCode, $segmentKey);
+                if (false === $locator) {
+                    throw new ResourceLocatorNotFoundException(\sprintf(
+                        'Couldn\'t generate resource locator for path: %s',
+                        $realPath->getPath()
+                    ));
```
Comment (alexander-schranz) [inline src/Sulu/Component/Content/Types/ResourceLocator/Mapper/PhpcrMapper.php:283]: @wachterjohannes are you fine with this change? See https://github.com/sulu/sulu/pull/7168/commits/d264cc1bfc910ef0e92372302596e198603cd485

Comment (mamazu) [reply src/Sulu/Component/Content/Types/ResourceLocator/Mapper/PhpcrMapper.php:283]: Before it would throw a type error, so it's a better error message.

Comment (wachterjohannes) [reply src/Sulu/Component/Content/Types/ResourceLocator/Mapper/PhpcrMapper.php:283]: @mamazu @alexander-schranz yes it makes sense to have that

**Pull request merged**